src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,413 @@
+                    at = at.substring(0, at.length() - 1);
+                    checkTaskDatetime(at);
+
+                    Event newTask = new Event(eventSplit[0], at);
+                    addExistingTask(savedTasks, newTask, taskStatus);
+                } else if (taskType == 'D') {
+                    String[] deadlineSplit = taskString.split(REGEX_DEADLINE_BRACKET_BY, 2);
+                    String by = processSplitString(deadlineSplit);
+                    by = by.substring(0, by.length() - 1);
+                    checkTaskDatetime(by);
+
+                    Deadline newTask = new Deadline(deadlineSplit[0], by);
+                    addExistingTask(savedTasks, newTask, taskStatus);
+                } else {
+                    throw new DukeException("Invalid saved task data found!");
+                }
+            }
+        } catch (FileNotFoundException | DukeException e) {
+            e.printStackTrace();
+        }
+
+        return savedTasks;
+    }
+
+    private static void addExistingTask(ArrayList<Task> taskList, Task newTask, char taskStatus) {
+        if (taskStatus == 'T') {
+            newTask.markAsDone();
+        }
+        taskList.add(newTask);
+    }
+
+    /**
+     * Checks if user input task number is an existing task.
+     *
+     * @param taskList        ArrayList containing tasks.
+     * @param taskDescription duke.task.Task index to be marked as done.
+     * @return True if task number is valid.
+     * @throws DukeException Throws exception if invalid task number.
+     */
+    private static boolean checkValidTaskNumber(ArrayList<Task> taskList, String taskDescription) throws DukeException {
+        // Checks if invalid done number is provided
+        if (taskDescription.equals("")) {
+            throw new DukeException(EXCEPTION_INVALID_TASK_NUMBER);
+        } else if (Integer.parseInt(taskDescription) > taskList.size()
+                || Integer.parseInt(taskDescription) <= 0) {
+            throw new DukeException(EXCEPTION_INVALID_TASK_NUMBER);
+        }
+
+        return true;
+    }
+
+    /**
+     * Marks an existing task as done.
+     *
+     * @param taskList        ArrayList containing tasks.
+     * @param taskDescription duke.task.Task index to be marked as done.
+     */
+    private static void processDoneCommand(ArrayList<Task> taskList, String taskDescription) throws DukeException {
+        if (checkValidTaskNumber(taskList, taskDescription)) {
+            Task tempTask = taskList.get(Integer.parseInt(taskDescription) - 1);
+
+            tempTask.markAsDone();
+
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println(" Nice! I've marked this task as done: ");
+            System.out.println("   " + tempTask);
+            System.out.println(HORIZONTAL_LINE + System.lineSeparator());
+            updateLocalList(taskList);
+        }
+    }
+
+    /**
+     * Delete specified task
```

Comment:
You may want to use Deletes instead of delete as present tense is used for method documentation